### PR TITLE
Amqp client notificacion

### PR DESCRIPTION
**Pull request by margaale**
Added the ability to send a notification to a message queue

### CODE REVIEW

File: doc/configuration.md [hunk 1]
```diff
@@ -173,7 +191,7 @@ providers:
   * `method`: HTTP method (default: `GET`). **required**
   * `headers`: Map of additional headers to be sent.
   * `timeout`: Timeout specifies a time limit for the request to be made. (default: `10`).
-
+
```
Comment (crazy-max) [inline doc/configuration.md:194]: Remove those spaces

File: doc/notifications.md [hunk 2]
```diff
@@ -1,12 +1,29 @@
...
 * [Mail](#mail)
 * [Rocket.Chat](#rocketchat)
 * [Slack](#slack)
 * [Telegram](#telegram)
 * [Webhook](#webhook)
 
+## Amqp
+
+You can send notifications any amqp compatible server, the body will be a JSON format that looks like:
+```json
```
Comment (crazy-max) [inline doc/notifications.md:13]: Add a new line after `like:`

File: internal/notif/amqp/client.go [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+		},
+	}
+}
+
+// Name returns notifier's name
+func (c *Client) Name() string {
+	return "amqp"
+}
+
+// Send creates and sends a slack notification with an entry
```
Comment (crazy-max) [inline internal/notif/amqp/client.go:38]: `slack` > `amqp`

File: internal/notif/amqp/client.go [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+func (c *Client) Send(entry model.NotifEntry) error {
+
+	username, err := utl.GetSecret(c.cfg.Username, c.cfg.UsernameFile)
+	if err != nil {
+		log.Err(err).Msg("Cannot retrieve username secret for amqp notifier")
+		return err
```
Comment (crazy-max) [inline internal/notif/amqp/client.go:44]: This will log the error twice. Remove `log.Err(err)...` line.

File: internal/notif/amqp/client.go [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+	}
+
+	password, err := utl.GetSecret(c.cfg.Password, c.cfg.PasswordFile)
+	if err != nil {
+		log.Err(err).Msg("Cannot retrieve password secret for amqp notifier")
+		return err
```
Comment (crazy-max) [inline internal/notif/amqp/client.go:50]: This will log the error twice. Remove `log.Err(err)...` line.

File: internal/notif/amqp/client.go [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+	}
+
+	connString := fmt.Sprintf("amqp://%s:%s@%s:%d/", username, password, c.cfg.Host, c.cfg.Port)
+
+	conn, err := amqp.Dial(connString)
+	failOnError(err, "Failed to connect to mq")
+
+	if err != nil {
+		return err
+	}
```
Comment (crazy-max) [inline internal/notif/amqp/client.go:60]: Do not use a dedicated function to handle errors:

```go
conn, err := amqp.Dial(connString)
if err != nil {
  return err
}
```

File: internal/notif/amqp/client.go [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+
+	defer conn.Close()
+
+	ch, err := conn.Channel()
+	failOnError(err, "Failed to open a channel")
+
+	if err != nil {
+		return err
+	}
```
Comment (crazy-max) [inline internal/notif/amqp/client.go:69]: Do not use a dedicated function to handle errors.

File: internal/notif/amqp/client.go [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+		false,       // delete when unused
+		false,       // exclusive
+		false,       // no-wait
+		nil,         // arguments
+	)
+	failOnError(err, "Failed to declare a queue")
+
+	if err != nil {
+		return err
+	}
```
Comment (crazy-max) [inline internal/notif/amqp/client.go:85]: Do not use a dedicated function to handle errors.

File: internal/notif/amqp/client.go [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+
+	body, err := buildBody(entry, c.app)
+	failOnError(err, "Failed build body")
+
+	if err != nil {
+		return err
+	}
```
Comment (crazy-max) [inline internal/notif/amqp/client.go:92]: Do not use a dedicated function to handle errors.

File: internal/notif/amqp/client.go [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+		q.Name,         // routing key
+		false,          // mandatory
+		false,          // immediate
+		amqp.Publishing{
+			ContentType: "application/json",
+			Body:        body,
+		})
+	failOnError(err, "Failed to publish a message")
+
+	return err
```
Comment (crazy-max) [inline internal/notif/amqp/client.go:105]: Use `return ch.Publish(` directly

File: internal/notif/amqp/client.go [hunk 11]
```diff
@@ -0,0 +1,142 @@
...
+		Created:      entry.Manifest.Created,
+		Architecture: entry.Manifest.Architecture,
+		Os:           entry.Manifest.Os,
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return body, nil
```
Comment (crazy-max) [inline internal/notif/amqp/client.go:135]: Use `return json.Marshal(struct {` directly.

File: internal/notif/amqp/client.go [hunk 12]
```diff
@@ -0,0 +1,142 @@
...
+}
+
+func failOnError(err error, msg string) {
+	if err != nil {
+		log.Err(err).Msg(msg)
+	}
+}
```
Comment (crazy-max) [inline internal/notif/amqp/client.go:142]: Remove this

File: doc/configuration.md [hunk 13]
```diff
@@ -129,6 +137,16 @@ providers:
 * `first_check_notif`: Send notification at the very first analysis of an image. _Optional_. (default: `false`).
 
 ### notif
+* `amqp`
```
Comment (crazy-max) [inline doc/configuration.md:140]: Add new line after `### notif`

Comment (crazy-max) [inline doc/configuration.md:140]: Add new line after `### notif`

**Comment by margaale** [comment]
Ja, Told you i was a newbie on go, thanks for the suggestion and the review

**Comment by margaale** [comment]
Hi? Did you have time to review this?

**Comment by crazy-max** [comment]
@margaale Will check this out soon ;)

**Comment by margaale** [comment]
Hey, do you want me to change the target to v3 and do a rebase?

**Comment by crazy-max** [comment]
@margaale That would be nice :)

**Comment by margaale** [comment]
Done

**Comment by crazy-max** [comment]
@margaale Ok, let's wait for the CI and I merge this PR!

**Comment by margaale** [comment]
ohhh, it failed, didn't see there was a change on Manifest...

**Comment by crazy-max** [comment]
@margaale 

> should i remove those properties?

`Os` and `Architecture` has been merged to `Platform`. So you have to replace them with `Platform` and don't forget the template. See [Slack notif](https://github.com/crazy-max/diun/blob/v3/internal/notif/slack/slack.go) as an example.

**Comment by margaale** [comment]
What template? is sending the status as a json property for the end user to decide which action to take

**Comment by crazy-max** [comment]
@margaale 

> What template?

Nevermind I thought the platform was included. Forget it.

File: doc/notifications.md [hunk 14]
```diff
@@ -1,12 +1,29 @@
...
 * [Gotify](#gotify)
 * [Mail](#mail)
 * [Rocket.Chat](#rocketchat)
 * [Slack](#slack)
 * [Telegram](#telegram)
 * [Webhook](#webhook)
 
+## Amqp
+
+You can send notifications any amqp compatible server, the body will be a JSON format that looks like:
```
Comment (crazy-max) [inline doc/notifications.md:12]: Spelling `notifications any` > `notifications to any`

**Pull request merged**